File: test/Bug/OffByOne.cs
using System;

public class Class1
{
	public Class1()
	{
        var things = new[]{1,2,3,};
        for (var i = 0; i <= things.Length; i++) {
            var thing = things[i];
        }

        //Better, do not catch
        foreach (var thing in things) {

        }
	}
}

File: test/Bug/OffByOne.php
<?
$a = [1,2,3,];

//Bug
for ($i=0; $i <= \count($a); $i++){}

//Better; do not catch.
for($i=0; $i < \count($a); $i++){}
for($i=0,$c=\count($a); $i<$c; $i++){}
foreach($a as $key => $value){}
\array_walk($a, static function(){});
\iterator_apply(new \ArrayIterator($a), static function(){});